Source/WebCore/html/GenericCachedHTMLCollection.cpp

```diff
@@ -74,10 +74,10 @@ bool GenericCachedHTMLCollection<traversalType>::elementMatches(Element& element
         return is<HTMLOptionElement>(element);
     case MapAreas:
         return element.hasTagName(areaTag);
-    case DocApplets:
-        return is<HTMLObjectElement>(element) && downcast<HTMLObjectElement>(element).containsJavaApplet();
     case DocEmbeds:
         return element.hasTagName(embedTag);
+    case DocEmpty:
```

Comment:
This seems very inefficient. As far as I can tell, this will traverse the whole DOM tree and return false for every element. We should not be traversing the DOM tree at all or even have any caching logic.

Reply:
Yeah, we probably want to introduce a new traversal type that's basically a no-op.

Reply:
If we introduced a new very simple `EmptyHTMLCollection` (similar to `EmptyNodeList`) which subclasses `HTMLCollection` and overrides:
```
Element* item(unsigned) const final { return nullptr; }
Element* namedItem(const AtomString& name) const final { return nullptr; }
unsigned length() const final { return 0; }
```

Then no traversal at all.

Reply:
Yeah, but we don't even need to subclass it. We can just create a new CollectionTraversal. Let me post a PR in a bit.

Reply:
Fixing this in https://bugs.webkit.org/show_bug.cgi?id=255389.